src/modules/item/sagas.ts

```diff
@@ -437,19 +437,10 @@ export function* itemSaga(legacyBuilder: LegacyBuilderAPI, builder: BuilderClien
     if (ItemModals.some(modal => openModals[modal])) {
       yield put(closeAllModals())
     } else if (openModals['CreateSingleItemModal']) {
-      if (location.pathname === locations.collections()) {
-        if (item.type === ItemType.EMOTE) {
-          // Redirect to the item editor
-          yield put(setItems([item]))
-          yield put(push(locations.itemEditor({ itemId: item.id })))
-        } else {
-          // Redirect to the newly created item details
-          yield put(push(locations.itemDetail(item.id)))
-        }
-      } else if (location.pathname === locations.collectionDetail(collectionId) && item.type === ItemType.EMOTE) {
+      if (location.pathname === locations.collectionDetail(collectionId) && item.type === ItemType.EMOTE) {
         // Redirect to the item editor
         yield put(setItems([item]))
-        yield put(push(locations.itemEditor({ collectionId, itemId: item.id, newItem: item.name })))
+        yield put(push(locations.itemEditor({ collectionId, itemId: item.id, newItem: item.name, fromParam: 'collections' })))
```

Comment:
What do you think if we send the `from` param as part of the router `state`? is the second parameter of the `push` method. It would be something like:

```
yield put(push(locations.itemEditor({ collectionId, itemId: item.id, newItem: item.name }), { fromParam: 'collections' }))
```